File: app/src/main/java/com/abstractclass/visitormanager/Globals.java
package com.abstractclass.visitormanager;

import android.Manifest;

public class Globals {

    // This is an arbitrary number we are using to keep track of the permission
    // request. Where an app has multiple context for requesting permission,
    // this can help differentiate the different contexts.
    public static final int REQUEST_CODE_PERMISSIONS = 10;


    // This is an array of all the permission specified in the manifest.
    public static String REQUIRED_PERMISSIONS = Manifest.permission.CAMERA;
}

File: app/src/main/java/com/abstractclass/visitormanager/Globals.kt
package com.abstractclass.visitormanager

import android.Manifest

class Globals {
    companion object {

        // This is an arbitrary number we are using to keep track of the permission
        // request. Where an app has multiple context for requesting permission,
        // this can help differentiate the different contexts.
        const val REQUEST_CODE_PERMISSIONS = 10

        // This is an array of all the permission specified in the manifest.
        val REQUIRED_PERMISSIONS = arrayOf(Manifest.permission.CAMERA,
                Manifest.permission.WRITE_EXTERNAL_STORAGE, Manifest.permission.READ_EXTERNAL_STORAGE)

        val FILE_READ_WRITE_PERMISSIONS: Array<String> = arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE,
                Manifest.permission.READ_EXTERNAL_STORAGE)
    }
}